src/apps/hw.js

```diff
@@ -70,20 +74,31 @@ export const streamAppInstall = (
         }
         if (e.type === "result") {
           // stream install with the result of list apps
-          const state = reducer(initState(e.result), {
-            type: "install",
-            name: appName,
-          });
-          if (isOutOfMemoryState(predictOptimisticState(state))) {
-            // it will not be possible to install so we will fallback to app-not-installed.
-            return of({ type: "app-not-installed", appName });
+          let state = initState(e.result);
+          appNames.forEach(
+            (name) => (state = reducer(state, { type: "install", name }))
+          );
```

Comment:
you can refactor this to:

```
const state = appNames.reduce((state, name) => reducer(state, { type: "install", name }), initState(e.result));
```

as it's a reducer :)

Reply:
nice, nice

Reply:
You got it